pan3d/explorers/slice_explorer.py

```diff
@@ -0,0 +1,495 @@
+from trame.app import get_server
+from trame.decorators import TrameApp, change
+from trame.ui.vuetify3 import SinglePageWithDrawerLayout
+from trame.widgets import vuetify3 as v3, vtk as vtkw, html
+from pan3d.dataset_builder import DatasetBuilder
+import vtk, numpy as np
+
+colors = [
+    "Rainbow",
+    "Inv Rainbow",
+    "Greyscale",
+    "Inv Greyscale",
+]
+
+
+class LookupTable:
+    Rainbow = 0
+    Inverted_Rainbow = 1
+    Greyscale = 2
+    Inverted_Greyscale = 3
+
+hsv_colors = {
+    "Rainbow" : {
+        "Hue" : (0.666, 0.0),
+        "Saturation" : (1.0, 1.0),
+        "Range" : (1.0, 1.0),
+    }, 
+    "Inv Rainbow" : {
+        "Hue" : (0.0, 0.666),
+        "Saturation" : (1.0, 1.0),
+        "Range" : (1.0, 1.0),
+    },
+    "Greyscale" : {
+        "Hue" : (0.0, 0.0),
+        "Saturation" : (0.0, 0.0),
+        "Range" : (0.0, 1.0),
+    },
+    "Inv Greyscale" : {
+        "Hue" : (0.0, 0.666),
+        "Saturation" : (0.0, 0.0),
+        "Range" : (1.0, 0.0),
+    },
+}
+
+def use_preset(sactor: vtk.vtkActor, dactor: vtk.vtkActor, preset: str) -> None:    
+    s_lut = sactor.GetMapper().GetLookupTable()
+    d_lut = dactor.GetMapper().GetLookupTable()
+    luts = [s_lut, d_lut]
+    for lut in luts:
+        hue = hsv_colors[preset]["Hue"] 
+        sat = hsv_colors[preset]["Saturation"]
+        rng = hsv_colors[preset]["Range"]
+        lut.SetHueRange(hue[0], hue[1])
+        lut.SetSaturationRange(sat[0], sat[1])
+        lut.SetValueRange(rng[0], rng[1])
+        lut.Build()      
+
+def update_preset(actor: vtk.vtkActor, logcale: bool) -> None:
+    lut = actor.GetMapper().GetLookupTable()
+    if logcale:
+        print("Setting log scale")
+        lut.SetScaleToLog10()
+    else:
+        print("Setting linear scale")
+        lut.SetScaleToLinear()
+    lut.Build()
+
+
+@TrameApp()
+class SliceExplorer:
+    def __init__(self, builder: DatasetBuilder = None, server=None):
+        self.server = get_server(server)
+        self.builder = builder
```

Comment:
what if builder is none? Can you create it here?

Reply:
Creating is the correct path is provided in the arguments to the app.